tests/Tests/Properties/Text.hs

```diff
@@ -38,18 +38,27 @@ tl_unstreamChunks x = f 11 x === f 1000 x
 tl_chunk_unchunk    = (TL.fromChunks . TL.toChunks) `eq` id
 tl_from_to_strict   = (TL.fromStrict . TL.toStrict) `eq` id
 
-
 s_map f           = map f  `eqP` (unpackS . S.map f)
 s_map_s f         = map f  `eqP` (unpackS . S.unstream . S.map f)
 sf_map p f        = (map f . L.filter p)  `eqP` (unpackS . S.map f . S.filter p)
+
 t_map f           = map f  `eqP` (unpackS . T.map f)
 tl_map f          = map f  `eqP` (unpackS . TL.map f)
+t_map_map f g     = (map f . map g) `eqP` (unpackS . T.map f . T.map g)
+tl_map_map f g    = (map f . map g)  `eqP` (unpackS . TL.map f . TL.map g)
+t_length_map f    = (L.length . map f)  `eqP` (T.length . T.map f)
+tl_length_map f   = (L.genericLength . map f)  `eqP` (TL.length . TL.map f)
+
 s_intercalate c   = (L.intercalate c . unSqrt) `eq`
                     (unpackS . S.intercalate (packS c) . map packS . unSqrt)
 t_intercalate c   = (L.intercalate c . unSqrt) `eq`
                     (unpackS . T.intercalate (packS c) . map packS . unSqrt)
 tl_intercalate c  = (L.intercalate c . unSqrt) `eq`
                     (unpackS . TL.intercalate (TL.pack c) . map TL.pack . unSqrt)
+t_length_intercalate c  = (L.length . L.intercalate c . unSqrt) `eq`
+                    (T.length . T.intercalate (packS c) . map packS . unSqrt)
+tl_length_intercalate c = (L.genericLength . L.intercalate c . unSqrt) `eq`
+                    (TL.length . TL.intercalate (TL.pack c) . map TL.pack . unSqrt)
```

Comment:
Is this not redundant with the tests without length? There may be some reason but I don't quite see it.

Reply:
Treat them as tests for `length`, not for `intercalate`. They are not immediately useful with current implementation of `length`, but I found them useful for other purposes, so it is easier to submit them separately.